optimum/exporters/onnx/base.py

```diff
@@ -572,7 +588,12 @@ def generate_dummy_inputs(self, framework: str = "pt", **kwargs):
                 )
 
         # refer to https://github.com/huggingface/optimum/pull/764
-        if self.use_past_in_inputs and "attention_mask" in dummy_inputs and self.PAD_ATTENTION_MASK_TO_PAST:
+        if (
+            self.use_past_in_inputs
+            and self.PAD_ATTENTION_MASK_TO_PAST
+            and self.use_cache_branch is not False
```

Comment:
Why do we need to add `self.use_cache_branch` in the conditional statement, isn't this method only used for exporting decoder and decoder with past ?

Reply:
We have two cases here:
* decoder with/without past separate, as `decoder_model.onnx` and `decoder_with_past_model.onnx`. In this case `self.use_cache_branch` is `None` and we indeed go into this controlflow when `self.use_past_in_inputs =True`, which is `decoder_with_past_model.onnx`.
* decoder with/without past merged, as `decoder_model_merged.onnx`. In this case, it is only when we use the cache branch in the merged model that we want to go into this controlflow, hence the need for the check `self.use_cache_branch is not False`.